llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7031,6 +7221,23 @@ void LoopVectorizationPlanner::plan(ElementCount UserVF, unsigned UserIC) {
   if (CM.foldTailByMasking())
     Legal->prepareToFoldTailByMasking();
 
+  for (auto Pair : CM.getPartialReductionChains()) {
+    // TODO: Allow creating partial reductions when predicating. The select at
+    // the end of the loop chooses between the phi value and most recent partial
+    // reduction result, both of which have different VFs to the active lane
+    // mask.
+    Instruction *Instr = Pair.first;
+    if (CM.blockNeedsPredicationForAnyReason(Instr->getParent())) {
+      LLVM_DEBUG(dbgs() << "LV: Removing the partial reduction for an "
+                           "instruction in a predicated block: "
+                        << *Instr << "\n");
+      PartialReductionsToRemove.insert(Instr);
+    }
+  }
```

Comment:
Currently, `collectValuesToIgnore` needs the partial reductions to exist, so we have to add them before that function is called. When we're able to remove the changes to `collectValuesToIgnore` I can remove this and only add the partial reductions if the block doesn't need predication. Unfortunately that predication information is only known after `collectValuesToIgnore` and `computeMaxVF` is called.